File: fsharp/darts/Darts.fs
module Darts

let score (x: double) (y: double): int = 
    match sqrt (x * x + y * y) with
    | s when s > 10.0 -> 0
    | s when s > 5.0 -> 1
    | s when s > 1.0 -> 5
    | _ -> 10

File: csharp/darts/Darts.cs
using System;

public static class Darts
{
    public static int Score(double x, double y)
    {
        const int outerRadius = 10;
        const int middleRadius = 5;
        const int innerRadius = 1;

        var radius = Math.Sqrt((x * x) + (y * y));

        if (radius > outerRadius) return 0;

        if (radius > middleRadius) return 1;

        if (radius > innerRadius) return 5;

        return 10;
    }
}
